src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/HeartbeatProvider.cs

```diff
@@ -0,0 +1,346 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Linq;
+    using System.Threading;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.ApplicationInsights.Extensibility;
+
+    /// <summary>
+    /// Implementation of health heartbeat functionality.
+    /// </summary>
+    internal class HeartbeatProvider : IDisposable, IHeartbeatProvider
+    {
+        /// <summary>
+        /// The default interval between heartbeats if not specified by the user. Left public for use in unit tests.
+        /// </summary>
+        public static int DefaultHeartbeatIntervalMs = 5000;
+
+        protected List<string> disabledDefaultFields; // string containing fields that are not to be sent with the payload. null means send everything available.
+
+        /// <summary>
+        /// The name of the health heartbeat metric item and operation context. 
+        /// </summary>
+        private static string heartbeatSyntheticMetricName = "HealthState";
+
+        /// <summary>
+        /// The SDK supplied 'default' payload items to send out with each health heartbeat.
+        /// </summary>
+        private ConcurrentDictionary<string, HeartbeatPropertyPayload> sdkPayloadItems;
+
+        /// <summary>
+        /// The extended payload items to send out with each health heartbeat.
+        /// </summary>
+        private ConcurrentDictionary<string, HeartbeatPropertyPayload> extendPayloadItems;
+
+        private bool disposedValue = false; // To detect redundant calls to dispose
+        private TimeSpan heartbeatInterval; // time between heartbeats emitted specified in milliseconds
+        private TelemetryClient telemetryClient; // client to use in sending our heartbeat
+        private int heartbeatsSent; // counter of all heartbeats
+
+        public HeartbeatProvider() : this(TimeSpan.FromMilliseconds(DefaultHeartbeatIntervalMs), null)
+        {
+        }
+
+        public HeartbeatProvider(TimeSpan heartbeatInterval, IEnumerable<string> disabledDefaultFields)
```

Comment:
who calls this constructor? is it for unit test?

Reply:
I wrote it for unit tests, but is no longer needed (we set and reset the interval & disabled properties by calling HeartbeatProvider.Initialize instead).

Removed the second constructor. Thanks for catching!